ocp_resources/catalog_source.py

```diff
@@ -1,34 +1,32 @@
-from ocp_resources.constants import TIMEOUT_4MINUTES
 from ocp_resources.resource import NamespacedResource
 
 
 class CatalogSource(NamespacedResource):
+    """
+    https://olm.operatorframework.io/docs/concepts/crds/catalogsource/
+    """
+
     api_group = NamespacedResource.ApiGroup.OPERATORS_COREOS_COM
 
     def __init__(
         self,
-        name=None,
-        namespace=None,
-        client=None,
         source_type=None,
         image=None,
         display_name=None,
         publisher=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
         update_strategy_registry_poll_interval=None,
         **kwargs,
     ):
-        super().__init__(
-            name=name,
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            source_type (str): Name of the source type.
+            image (str): Image index for the catalog.
+            display_name (str): Display name for the catalog in the web console and CLI.
+            publisher (str): Name of the publisher.
```

Comment:
should raise if any of these are missing

Reply:
@rnetser Every pr is pointing at an issue like:
#1414 (Add validation for mandatory args for class constructor in 'catalog_source.py')

Can I fix it in this PR? (will change the pr title)
More resources will be updated similarly.

Reply:
i think that if you're already doing the job of adding docstrings and mapping what is mandatory, it will be easier to add the checks in the same pr; wdyt?